pkg/services/ngalert/state/cache.go

```diff
@@ -38,6 +39,15 @@ func (c *cache) getOrCreate(ctx context.Context, alertRule *ngModels.AlertRule,
 
 	ruleLabels, annotations := c.expandRuleLabelsAndAnnotations(ctx, alertRule, result, extraLabels)
 
+	values := make(map[string]float64)
+	for _, v := range result.Values {
+		if v.Value != nil {
+			values[v.Var] = *v.Value
```

Comment:
Just for my understanding, `v.Var` is the name of the stage, right? So i.e. "C" or "A" or whatever you call it.

Reply:
That's right! 😄